python-sdk/pachyderm_sdk/config.py

```diff
@@ -12,20 +14,65 @@
 class ConfigFile:
     """A parsed Pachyderm config file."""
 
-    def __init__(self, config_file: Union[Path, str]):
+    def __init__(self, data: dict):
         """
+        Parameters
+        ----------
+        data : dict (JSON)
+            The config file data. The only required fields are:
+            {
+              "v2": {
+                "active_context": ...,
+                "contexts": {...}
+              }
+            }
+        """
+        if (v2 := data.get("v2")) is None or not (
+            "active_context" in v2 and "contexts" in v2
+        ):
+            raise ValueError(
+                'Expected format: {"v2": {"active_context": ..., "contexts": {...}}'
```

Comment:
nit: maybe `Expected format` is not the best term to use, since having `user_id` at the top level is also valid. perhaps something more like "missing required argument v2.active_context/v2.contexts" or something